src/linalg/decomposition.rs

```diff
@@ -86,12 +87,34 @@ impl<T: ComplexField, R: Dim, C: Dim, S: Storage<T, R, C>> Matrix<T, R, C, S> {
             + Allocator<T, R, DimMinimum<R, C>>
             + Allocator<T, DimMinimum<R, C>>
             + Allocator<T::RealField, DimMinimum<R, C>>
-            + Allocator<T::RealField, DimDiff<DimMinimum<R, C>, U1>>,
+            + Allocator<T::RealField, DimDiff<DimMinimum<R, C>, U1>>
+            + Allocator<(usize, usize), DimMinimum<R, C>>
+            + Allocator<(T::RealField, usize), DimMinimum<R, C>>,
     {
         SVD::new(self.into_owned(), compute_u, compute_v)
     }
 
+    /// Computes the Singular Value Decomposition using implicit shift.
+    /// nalgebra does not guarantee any particular order of the singular values by default.
```

Comment:
Let’s make the comment similar to `svd`:
```suggestion
    /// The singular values are not guaranteed to be sorted in any particular order.
    /// If a descending order is required, consider using `svd` instead.
```